src/slider.js

```diff
@@ -61,7 +63,7 @@ var Slider = React.createClass({
     var newProps;
     if (this.state.breakpoint) {
       newProps = this.props.responsive.filter(resp => resp.breakpoint === this.state.breakpoint);
-      settings = newProps[0].settings === 'unslick' ? 'unslick' : assign({}, this.props, newProps[0].settings);
+      settings = newProps[0].settings === 'unslick' ? 'unslick' : assign({}, defaultProps, this.props, newProps[0].settings);
```

Comment:
Here `defaultProps` has never been used but if you look at non-responsive version, they are used:

```javascript
settings = assign({}, defaultProps, this.props);
```